mobile/src/main/java/rs/readahead/washington/mobile/views/fragment/MicFragment.kt

```diff
@@ -1,17 +1,80 @@
-import androidx.fragment.app.Fragment
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
+import android.widget.ImageButton
+import android.widget.ImageView
+import android.widget.TextView
+import androidx.appcompat.app.AlertDialog
+import androidx.appcompat.content.res.AppCompatResources
+import androidx.core.app.ActivityCompat
+import com.hzontal.tella_vault.VaultFile
+import io.reactivex.disposables.CompositeDisposable
+import org.hzontal.shared_ui.bottomsheet.BottomSheetUtils
 import rs.readahead.washington.mobile.R
+import rs.readahead.washington.mobile.media.AudioRecorder
+import rs.readahead.washington.mobile.media.AudioRecorder.AudioRecordInterface
+import rs.readahead.washington.mobile.media.MediaFileHandler
+import rs.readahead.washington.mobile.mvp.contract.IAudioCapturePresenterContract
+import rs.readahead.washington.mobile.mvp.contract.IMetadataAttachPresenterContract
+import rs.readahead.washington.mobile.mvp.contract.ITellaFileUploadSchedulePresenterContract
+import rs.readahead.washington.mobile.mvp.presenter.AudioCapturePresenter
+import rs.readahead.washington.mobile.mvp.presenter.MetadataAttacher
+import rs.readahead.washington.mobile.mvp.presenter.TellaFileUploadSchedulePresenter
+import rs.readahead.washington.mobile.util.DateUtil.getDateTimeString
+import rs.readahead.washington.mobile.util.StringUtils
+import rs.readahead.washington.mobile.views.base_ui.MetadataBaseLockFragment
+import timber.log.Timber
+
+import java.util.*
+import java.util.concurrent.TimeUnit
 
 private const val ARG_PARAM1 = "param1"
 private const val ARG_PARAM2 = "param2"
 
-class MicFragment : Fragment() {
+const val RECORD_REQUEST_CODE = 2001
+
+class MicFragment : MetadataBaseLockFragment(), AudioRecordInterface,
+    IAudioCapturePresenterContract.IView,
+    ITellaFileUploadSchedulePresenterContract.IView,
+    IMetadataAttachPresenterContract.IView {
+
     // TODO: Rename and change types of parameters
+    private val TIME_FORMAT: String = "%02d:%02d"
+    //var RECORDER_MODE = "rm"
+
+    private var animator: ObjectAnimator? = null
+
+    private var notRecording = false
+
+    private val UPDATE_SPACE_TIME_MS: Long = 60000
+    private var lastUpdateTime: Long = 0
+
+    // handling MediaFile
+    private var handlingMediaFile: VaultFile? = null
+
+    // recording
+    private var uploadPresenter: TellaFileUploadSchedulePresenter? = null
+    private val disposable = CompositeDisposable()
+    var presenter: AudioCapturePresenter? = null
+    var audioRecorder: AudioRecorder? = null
+
+    lateinit var metadataAttacher: MetadataAttacher
+    lateinit var mRecord: ImageButton
+    lateinit var mPlay: ImageButton
+    lateinit var mStop: ImageButton
+    lateinit var mTimer: TextView
+    lateinit var freeSpace: TextView
+    lateinit var redDot: ImageView
+    lateinit var recordingName: TextView
+
     private var param1: String? = null
     private var param2: String? = null
```

Comment:
what is the use of this params ?

Reply:
True, don't need it. In fact, I just need one for animation.